common/djangoapps/third_party_auth/pipeline.py

```diff
@@ -875,6 +875,14 @@ def user_details_force_sync(auth_entry, strategy, details, user=None, *args, **k
                 changed[provider_field] = current_value
                 setattr(model, field, provider_value)
 
+        # Generate fullname only for IES IDP.
+        if (not details.get('fullname') and details.get('first_name') and details.get('last_name') and
+            ('iam-stage.pearson.com' in current_provider.entity_id or 'iam.pearson.com' in current_provider.entity_id)):
```

Comment:
To check the last or condition, I would suggest to use:
```suggestion
            any(['iam-stage.pearson.com' in current_provider.entity_id, 'iam.pearson.com' in current_provider.entity_id])):
```

Reply:
Taking this in consideration.